File: 4412-count-good-cyclic-rotations/count-good-cyclic-rotations.cpp
class Solution {
public:
    int countGoodRotations(vector<int>& nums) {
        int n = nums.size();
        long long total = 0;
        long long left = 0;
        for(int it: nums){
            total+=it;
        }
        for(int i = 0; i<n/2; i++){
            left+=nums[i];
        }
        int ans = 0;
        for(int k = 0; k<n; k++){
            if(2*left>total){
                ans++;
            }
            left-=nums[k];
            left+=nums[(k+n/2)%n];
        }
        return ans;
    }
};

File: 4412-count-good-cyclic-rotations/count-good-cyclic-rotations.py
class Solution:
    def countGoodRotations(self, nums: list[int]) -> int:
        n = len(nums)
        total = sum(nums)
        l = 0
        r = n//2-1
        left = sum(nums[l:r+1])
        ans = 0
        for _ in range(n):
            if(2*left>total):
                ans+=1
            left-=nums[l]
            l = (l+1)%n
            r = (r+1)%n
            left+=nums[r]
        return ans